### PR TITLE
count tiledImage._tilesLoading

### PR DESCRIPTION
**Pull request by thejohnhoffer**
TL;DR: I've added a `tiledImage._tilesLoading` flag that tracks the requests for tiles that are still pending have not yet received data. This allows `tiledImage._fullyLoaded` to behave as expected.

The `tiledImage._fullyLoaded` term [pulled into master as request 837](https://github.com/openseadragon/openseadragon/pull/837) as well as the associated tiledImage event `'fully-loaded-change'` should allow the developer to design viewers that respond to the **completed loading of all tiles** for a current view in a `tiledImage`.

> TiledImage now has a getFullyLoaded() function that tells you whether all tiles necessary for this TiledImage to draw at the current view have been loaded. You can watch this status as well by listening to the fully-loaded-change event on the TiledImage.

-- @iangilman [2016-08-10](https://github.com/openseadragon/openseadragon/pull/837#issuecomment-238943723)

As the current master branch stands, the `tiledImage._fullLoaded` flag actually only informs the developer whether all tiles have _begun loading._ The changes here make `tiledImage._fullLoaded`, `tiledImage.getFullyLoaded`, and the `'fully-loaded-change'` event inform the developer whether all tiles have _finished loading._

Many thanks to Ian Gilman for walking me through the changes needed to make this happen.

### CODE REVIEW

**Comment by thejohnhoffer** [comment]
Ah- it just occurred to me that this will need to be reset on calling tiledImage.reset

**Comment by iangilman** [comment]
Looks great! Thank you for this fix :-)
